src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/MigrateVisitService.kt

```diff
@@ -95,24 +102,66 @@ class MigrateVisitService(
     return visitEntity.reference
   }
 
-  private fun sendTelemetryData(visitEntity: Visit) {
-    telemetryClient.trackEvent(
-      TelemetryVisitEvents.VISIT_MIGRATED_EVENT.eventName,
-      mapOf(
-        "reference" to visitEntity.reference,
-        "prisonerId" to visitEntity.prisonerId,
-        "prisonId" to visitEntity.prison.code,
-        "visitType" to visitEntity.visitType.name,
-        "visitRoom" to visitEntity.visitRoom,
-        "visitRestriction" to visitEntity.visitRestriction.name,
-        "visitStart" to visitEntity.visitStart.toString(),
-        "visitStatus" to visitEntity.visitStatus.name,
-        "outcomeStatus" to visitEntity.outcomeStatus?.name,
-      ),
-      null,
+  fun cancelVisit(reference: String, cancelVisitDto: CancelVisitDto): VisitDto {
+    val cancelOutcome = cancelVisitDto.cancelOutcome
+
+    if (visitRepository.isBookingCancelled(reference)) {
+      // If already canceled then just return object and do nothing more!
+      VisitService.LOG.debug("The visit $reference has already been canceled!")
+      val canceledVisit = visitRepository.findByReference(reference)!!
+      return VisitDto(canceledVisit)
+    }
+
+    val visitEntity = visitRepository.findBookedVisit(reference) ?: throw VisitNotFoundException("Canceled migrated visit $reference not found ")
+
+    visitEntity.visitStatus = CANCELLED
+    visitEntity.outcomeStatus = cancelOutcome.outcomeStatus
+    visitEntity.cancelledBy = cancelVisitDto.actionedBy
+
+    cancelOutcome.text?.let {
+      visitEntity.visitNotes.add(createVisitNote(visitEntity, VisitNoteType.VISIT_OUTCOMES, cancelOutcome.text))
+    }
+
+    sendMigratedTrackEvent(visitEntity, TelemetryVisitEvents.CANCELLED_VISIT_MIGRATED_EVENT)
```

Comment:
@lbennett-moj This is were we differentiate between synchronised and migrated
`CANCELLED_VISIT_MIGRATED_EVENT`